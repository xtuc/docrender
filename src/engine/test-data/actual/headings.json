{
  "title": "Publish-Bot test - headings",
  "body": {
    "content": [
      {
        "endIndex": 1,
        "sectionBreak": {
          "sectionStyle": {
            "columnSeparatorStyle": "NONE",
            "contentDirection": "LEFT_TO_RIGHT",
            "sectionType": "CONTINUOUS"
          }
        }
      },
      {
        "startIndex": 1,
        "endIndex": 7,
        "paragraph": {
          "elements": [
            {
              "startIndex": 1,
              "endIndex": 7,
              "textRun": {
                "content": "Title\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.640doajndp7y",
            "namedStyleType": "TITLE",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 7,
        "endIndex": 16,
        "paragraph": {
          "elements": [
            {
              "startIndex": 7,
              "endIndex": 16,
              "textRun": {
                "content": "subtitle\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.un3dy9grg5fl",
            "namedStyleType": "SUBTITLE",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 16,
        "endIndex": 26,
        "paragraph": {
          "elements": [
            {
              "startIndex": 16,
              "endIndex": 26,
              "textRun": {
                "content": "Heading 1\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.n63wo9hzskvu",
            "namedStyleType": "HEADING_1",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 26,
        "endIndex": 36,
        "paragraph": {
          "elements": [
            {
              "startIndex": 26,
              "endIndex": 36,
              "textRun": {
                "content": "Heading 2\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.vvg5j1raj2sy",
            "namedStyleType": "HEADING_2",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 36,
        "endIndex": 50,
        "paragraph": {
          "elements": [
            {
              "startIndex": 36,
              "endIndex": 50,
              "textRun": {
                "content": "Heading 1 - 2\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.43t62nfsnfkz",
            "namedStyleType": "HEADING_1",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 50,
        "endIndex": 64,
        "paragraph": {
          "elements": [
            {
              "startIndex": 50,
              "endIndex": 64,
              "textRun": {
                "content": "Heading 2 - 2\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.83zrp6j2ychh",
            "namedStyleType": "HEADING_2",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 64,
        "endIndex": 78,
        "paragraph": {
          "elements": [
            {
              "startIndex": 64,
              "endIndex": 78,
              "textRun": {
                "content": "Heading 3 - 2\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.aus8w8t5kf46",
            "namedStyleType": "HEADING_3",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 78,
        "endIndex": 88,
        "paragraph": {
          "elements": [
            {
              "startIndex": 78,
              "endIndex": 88,
              "textRun": {
                "content": "Heading 3\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.j45i1jad14f8",
            "namedStyleType": "HEADING_3",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 88,
        "endIndex": 98,
        "paragraph": {
          "elements": [
            {
              "startIndex": 88,
              "endIndex": 98,
              "textRun": {
                "content": "Heading 4\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.nqzaza4b3ac8",
            "namedStyleType": "HEADING_4",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 98,
        "endIndex": 112,
        "paragraph": {
          "elements": [
            {
              "startIndex": 98,
              "endIndex": 112,
              "textRun": {
                "content": "Heading 1 two\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.fdm1acb4qkz8",
            "namedStyleType": "HEADING_1",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 112,
        "endIndex": 118,
        "paragraph": {
          "elements": [
            {
              "startIndex": 112,
              "endIndex": 118,
              "textRun": {
                "content": "lines\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "headingId": "h.ov3jc2e8b9ve",
            "namedStyleType": "HEADING_1",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      }
    ]
  },
  "documentStyle": {
    "background": {
      "color": {}
    },
    "pageNumberStart": 1,
    "marginTop": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginBottom": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginRight": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginLeft": {
      "magnitude": 72,
      "unit": "PT"
    },
    "pageSize": {
      "height": {
        "magnitude": 841.8897637795277,
        "unit": "PT"
      },
      "width": {
        "magnitude": 595.27559055118115,
        "unit": "PT"
      }
    },
    "marginHeader": {
      "magnitude": 36,
      "unit": "PT"
    },
    "marginFooter": {
      "magnitude": 36,
      "unit": "PT"
    },
    "useCustomHeaderFooterMargins": true
  },
  "namedStyles": {
    "styles": [
      {
        "namedStyleType": "NORMAL_TEXT",
        "textStyle": {
          "bold": false,
          "italic": false,
          "underline": false,
          "strikethrough": false,
          "smallCaps": false,
          "backgroundColor": {},
          "foregroundColor": {
            "color": {
              "rgbColor": {}
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          },
          "baselineOffset": "NONE"
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "alignment": "START",
          "lineSpacing": 115,
          "direction": "LEFT_TO_RIGHT",
          "spacingMode": "COLLAPSE_LISTS",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "unit": "PT"
          },
          "borderBetween": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderTop": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderBottom": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderLeft": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderRight": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          },
          "indentEnd": {
            "unit": "PT"
          },
          "keepLinesTogether": false,
          "keepWithNext": false,
          "avoidWidowAndOrphan": true,
          "shading": {
            "backgroundColor": {}
          },
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_1",
        "textStyle": {
          "fontSize": {
            "magnitude": 20,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 20,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_2",
        "textStyle": {
          "bold": false,
          "fontSize": {
            "magnitude": 16,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 18,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_3",
        "textStyle": {
          "bold": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.2627451,
                "green": 0.2627451,
                "blue": 0.2627451
              }
            }
          },
          "fontSize": {
            "magnitude": 14,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 16,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_4",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 12,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 14,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_5",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_6",
        "textStyle": {
          "italic": true,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "TITLE",
        "textStyle": {
          "fontSize": {
            "magnitude": 26,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 3,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "SUBTITLE",
        "textStyle": {
          "italic": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 15,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 16,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      }
    ]
  },
  "revisionId": "ALBJ4Lsxa7KncmjZ5mAJlJE96tNgtPfEhq3M8gd4UdCv2BdllGXmh91Y9IBmfJjRjKj3Ec7zTb0zYaKUcmVR_wM",
  "suggestionsViewMode": "SUGGESTIONS_INLINE",
  "documentId": "1xnsCkUbfN8-oF4pmW-k7AGJJlo9DIorCHcSPeJ9zKM0"
}
