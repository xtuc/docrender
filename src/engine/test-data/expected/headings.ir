(
    content: [
        Clear,
        Title((
            level: 1,
            text: Text("Title"),
        ), []),
        Clear,
        Paragraph([
            InlineText([
                Text("subtitle"),
            ], true),
        ]),
        Clear,
        Title((
            level: 2,
            text: Text("Heading 1"),
        ), []),
        Clear,
        Title((
            level: 3,
            text: Text("Heading 2"),
        ), []),
        Clear,
        Title((
            level: 2,
            text: Text("Heading 1 - 2"),
        ), []),
        Clear,
        Title((
            level: 3,
            text: Text("Heading 2 - 2"),
        ), []),
        Clear,
        Title((
            level: 4,
            text: Text("Heading 3 - 2"),
        ), []),
        Clear,
        Title((
            level: 4,
            text: Text("Heading 3"),
        ), []),
        Clear,
        Title((
            level: 5,
            text: Text("Heading 4"),
        ), []),
        Clear,
        Title((
            level: 2,
            text: Text("Heading 1 two"),
        ), []),
        Clear,
        Title((
            level: 2,
            text: Text("lines"),
        ), []),
        Clear,
    ],
    styles: {
        "pbNORMAL_TEXT": [
            ("line-height", "1.7249999"),
        ],
    },
)