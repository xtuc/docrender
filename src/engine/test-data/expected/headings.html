<style>
  .pbNORMAL_TEXT {
    line-height: 1.7249999;
  }
</style>
<h1 id="pb-Title">Title</h1>
<p>subtitle</p>
<h2 id="pb-Heading%201">Heading 1</h2>
<h3 id="pb-Heading%202">Heading 2</h3>
<h2 id="pb-Heading%201%20-%202">Heading 1 - 2</h2>
<h3 id="pb-Heading%202%20-%202">Heading 2 - 2</h3>
<h4 id="pb-Heading%203%20-%202">Heading 3 - 2</h4>
<h4 id="pb-Heading%203">Heading 3</h4>
<h5 id="pb-Heading%204">Heading 4</h5>
<h2 id="pb-Heading%201%20two">Heading 1 two</h2>
<h2 id="pb-lines">lines</h2>
